Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -336,9 +338,46 @@ public SecurityChanges ApplyUniverseSelection(Universe universe, DateTime dateTi
         /// </summary>
         /// <param name="utcStart">The current date time in utc</param>
         /// <returns>Will return true if any subscription was added</returns>
-        public bool AddPendingCurrencyDataFeeds(DateTime utcStart)
+        public bool AddPendingInternalDataFeeds(DateTime utcStart)
         {
             var added = false;
+            if (!_initializedSecurityBenchmark)
+            {
+                _initializedSecurityBenchmark = true;
+
+                var securityBenchmark = _algorithm.Benchmark as SecurityBenchmark;
+                if (securityBenchmark != null)
```

Comment:
I wonder if there's a better, more encapsulated way to perform this. This, upon first glance, appears to be leaking details  of the security benchmark into the universe selection. Let's chat about the reasoning behind placing this logic here and perhaps some other potential options for performing this initializing behavior.

Reply:
Agree, this is the sticky point of this PR 😅 . Part of the reason I place it here is that this is where we are detecting/adding the internal currency subscriptions - But agree, I believe both additions could be handled differently - I guess what we are missing is an owner/responsible for the internal subscriptions altogether